x-pack/plugins/enterprise_search/server/collectors/app_search/telemetry.test.ts

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { savedObjectsRepositoryMock } from 'src/core/server/mocks';
+
+import { registerTelemetryUsageCollector, incrementUICounter } from './telemetry';
+
+/**
+ * Since these route callbacks are so thin, these serve simply as integration tests
+ * to ensure they're wired up to the lib functions correctly. Business logic is tested
+ * more thoroughly in the lib/telemetry tests.
+ */
+describe('App Search Telemetry Usage Collector', () => {
+  const makeUsageCollectorStub = jest.fn();
+  const registerStub = jest.fn();
+
+  const savedObjectsRepoStub = {
+    get: () => ({
+      attributes: {
+        'ui_viewed.setup_guide': 10,
+        'ui_viewed.engines_overview': 20,
+        'ui_error.cannot_connect': 3,
+        'ui_error.no_as_account': 4,
+        'ui_clicked.header_launch_button': 50,
+        'ui_clicked.engine_table_link': 60,
+      },
+    }),
+    incrementCounter: jest.fn(),
+  };
```

Comment:
Note to self - if we weren't specifically returning a mocked response in get, we could also use  Kibana's `savedObjectsRepositoryMock` test helper (via `import { savedObjectsRepositoryMock } from 'src/core/server/mocks';`, but it was as easy or faster to just stub out what we needed here)